apps/emqx/src/emqx_config.erl

```diff
@@ -238,7 +248,22 @@ erase(RootName) ->
     persistent_term:erase(?PERSIS_KEY(?RAW_CONF, bin(RootName))).
 
 -spec put(emqx_map_lib:config_key_path(), term()) -> ok.
-put(KeyPath, Config) -> do_put(?CONF, KeyPath, Config).
+put(KeyPath, Config) ->
+    Putter = fun(Path, Map, Value) ->
+        emqx_map_lib:deep_put(Path, Map, Value)
+    end,
+    do_put(?CONF, safe, Putter, KeyPath, Config).
+
+-spec force_put(emqx_map_lib:config_key_path(), term()) -> ok.
```

Comment:
Would be nice to have a comment explaining the difference with `put`, since it's not immediately obvious.